ash/src/util.rs

```diff
@@ -28,10 +30,9 @@ pub struct AlignIter<'a, T: 'a> {
 
 impl<T: Copy> Align<T> {
     pub fn copy_from_slice(&mut self, slice: &[T]) {
-        use std::slice::from_raw_parts_mut;
         if self.elem_size == size_of::<T>() as u64 {
             unsafe {
-                let mapped_slice = from_raw_parts_mut(self.ptr.cast(), slice.len());
+                let mapped_slice = slice::from_raw_parts_mut(self.ptr as *mut T, slice.len());
```

Comment:
We just cleaned these loose `as` casts up :angry:

Reply:
I recall some clippy or rust lint exists to prevent these from creeping back in?

Reply:
Perhaps [`ptr_as_ptr`](https://rust-lang.github.io/rust-clippy/master/#ptr_as_ptr) might help us with this but I don't recall seriously looking into it via #685. With that, would you still be able to implement this using `core`? Since this cast function is available on the `core` primitive pointer type?

Reply:
I think we cannot use this because there are a few `p_next` variants that store `*const T` in a `*mut T`, and `cast_mut()` hasn't been stable for long enough.